Assets/MixedRealityToolkit/Providers/BaseInputDeviceManager.cs

```diff
@@ -61,36 +183,168 @@ public abstract class BaseInputDeviceManager : BaseDataProvider<IMixedRealityInp
         /// <param name="useSpecificType">Only register pointers with a specific type.</param>
         protected virtual IMixedRealityPointer[] RequestPointers(SupportedControllerType controllerType, Handedness controllingHand)
         {
-            var pointers = new List<IMixedRealityPointer>();
-
-            if ((Service != null) &&
-                (InputSystemProfile != null) &&
-                InputSystemProfile.PointerProfile != null)
+            using (RequestPointersPerfMarker.Auto())
```

Comment:
Do you want to leave this in?

Reply:
yes see other comment. Markers are conditionally compiled out for release builds

Reply:
The docs only mention:

>Methods Begin and End are marked with ConditionalAttribute

Looks like we might be allocating an `Auto` that then does nothing because its internals are compiled out:

>Begin is called in the constructor and End in the Dispose method.

Though it's a struct, so the perf impacts may be minimal in this case

Reply:
@keveleigh  True indeed. I know it's every frame, but the two key pieces for me of having minimal impact
1) it's a struct, so on the stack and not on the heap. Thus no GC, basically like allocating some primitives in a function
2) It does not execute begin/end() so computation is does nothing but be allocated on the stack. 

I went with the auto because I like the scoping nature of it. Of course, in the future, probably best practice will be to do the following so get best of both worlds

```csharp
ProfileMarker.Begin();
{
....
}
ProfileMarker.End();
```